src/endpoints/substrate-endpoint/services/service-request.service.ts

```diff
@@ -64,114 +73,91 @@ export class ServiceRequestService {
       const oneDbioEqualToUsd = exchangeBalance?.dbioToUsd || undefined;
 
       // Accumulate totalRequests and totalValue by country
-      const requestByCountryDict = {};
+      const requestByCountryDict: {
+        [country: string]: RequestByCountryDictInterface;
+      } = {};
       for (const req of hits) {
         const {
-          _source: { request },
+          _source: { country, service_request },
         } = req;
 
-        if (!requestByCountryDict[request.country]) {
-          requestByCountryDict[request.country] = {
+        if (!requestByCountryDict[country]) {
+          requestByCountryDict[country] = {
             totalRequests: 0,
             totalValue: 0,
             services: {},
           };
-
-          endIndex++;
         }
+        for (const service of service_request) {
+          const value = Number(service.amount.split(',').join('')) / 10 ** 18;
+          requestByCountryDict[country].totalRequests += 1;
+          const currValueByCountry = Number(
+            requestByCountryDict[country].totalValue,
+          );
+          requestByCountryDict[country].totalValue = currValueByCountry + value;
 
-        const value =
-          Number(request.staking_amount.split(',').join('')) / 10 ** 18;
-        requestByCountryDict[request.country].totalRequests += 1;
-        const currValueByCountry = Number(
-          requestByCountryDict[request.country].totalValue,
-        );
-        requestByCountryDict[request.country].totalValue =
-          currValueByCountry + value;
+          if (
+            !requestByCountryDict[country]['services'][
+              service.region + '-' + service.city + '-' + service.category
+            ]
+          ) {
+            requestByCountryDict[country]['services'][
+              service.region + '-' + service.city + '-' + service.category
+            ] = {
+              category: service.category,
+              regionCode: service.region,
+              city: service.city,
+              totalRequests: 0,
+              totalValue: {
+                dbio: 0,
+                dai: 0,
+                usd: 0,
+              },
+            };
+          }
 
-        if (
-          !requestByCountryDict[request.country]['services'][
-            request.region + '-' + request.city + '-' + request.service_category
-          ]
-        ) {
-          requestByCountryDict[request.country]['services'][
-            request.region + '-' + request.city + '-' + request.service_category
-          ] = {
-            category: request.service_category,
-            regionCode: request.region,
-            city: request.city,
-            totalRequests: 0,
-            totalValue: {
-              dbio: 0,
-              dai: 0,
-              usd: 0,
-            },
-          };
+          requestByCountryDict[country]['services'][
+            service.region + '-' + service.city + '-' + service.category
```

Comment:
Please turn this into its own variable somewhere to reuse.

Reply:
done